File: app/src/main/java/co/lateralview/myapp/domain/repository/implementation/ParserManagerImpl.java
package co.lateralview.myapp.domain.repository.implementation;

import com.google.gson.Gson;

import java.lang.reflect.Type;

import co.lateralview.myapp.domain.repository.interfaces.ParserManager;

public class ParserManagerImpl implements ParserManager {
    private Gson mGson;

    public ParserManagerImpl(Gson gson) {
        mGson = gson;
    }

    public String toJson(Object object) {
        return mGson.toJson(object);
    }

    public <T> T fromJson(String json, Class<T> type) {
        return mGson.fromJson(json, type);
    }

    @Override
    public <T> T fromJson(String json, Type type) {
        return mGson.fromJson(json, type);
    }
}

File: app/src/main/java/co/lateralview/myapp/domain/repository/implementation/ParserManagerImpl.kt
package co.lateralview.myapp.domain.repository.implementation

import co.lateralview.myapp.domain.repository.interfaces.ParserManager
import com.google.gson.Gson
import java.lang.reflect.Type

class ParserManagerImpl(private val gson: Gson) : ParserManager {

    override fun toJson(`object`: Any): String {
        return gson.toJson(`object`)
    }

    override fun <T> fromJson(json: String, type: Class<T>): T {
        return gson.fromJson(json, type)
    }

    override fun <T> fromJson(json: String, type: Type): T {
        return gson.fromJson<T>(json, type)
    }
}